routes/adminapi.js

```diff
@@ -84,6 +87,122 @@ router.post('/saveword', function(req, res) {
   });
 });
 
+/*
+curl -H "Content-Type: application/json" -H "Accept: application/json" \
+-X POST \
+-H "Cookie: userid=123" \
+-d '{"words": [{"word":"dfg1","images":["URL1", "URL2"]}, {"word":"ghj","images":["URL1", "URL2"]}]}' http://localhost:8080/adminapi/saveword
+*/
+router.post('/bulksave', function(req, res) {
+  if (!(config.COOKIE_NAME in req.cookies))
+  return res.status(400).send('Cookie not provided.');
+  const sessionId = req.cookies[config.COOKIE_NAME];
+  db.getUser(sessionId, (user) => {
+    if (user === false || user === 'No User.')
+      return res.status(500).send('Error Occurred in user lookup');
+    if (config.WHITELISTED_ADMINS.indexOf(user.email) < 0)
+      return res.status(400).send('Invalid user');
+
+    let dbPromises = [];
+    if (req.body.hasOwnProperty('word')) {
+      dbPromises.push(saveWord(req.body, user.email));
+    } else {
+      res.status(400).send('Invalid request format');
+    }
+
+    Promise.all(dbPromises).then((results) => {
+      let response = {Saved: 0, Duplicate: 0, errors: 0};
+      for (var i = 0; i < results.length; i++) {
+        if (results[i] === 'Duplicate' || results[i] === 'Saved') {
+          response[results[i]] += 1;
+        } else response.errors += 1;
+      }
+      res.send(JSON.stringify(response));
+    })
+    .catch((err) => {
+      res.status(500).send('Unknown Error Occurred, some words might have been saved');
+    });
+  });
+});
+
+function saveWord(payload, email) {
+  return new Promise((resolve, reject) => {
+    console.log(payload.word);
+    db.checkWordExists(payload.word, (err, message) => {
+      if (err) return reject(err);
+      if (message === true) {
+        return resolve('Duplicate');
+      }
+      payload.author = email;
+      db.insertWordInCollection(payload, (err, message) => {
+        if (err) return reject(err);
+        resolve(message);
+      });
+    });
+  });
+}
+
+// Endpoint to pick the words from ./routes/wordlist.txt and create game configs
+// curl localhost:8080/adminapi/makegames?wordcount=2
+router.get('/makegames', (req, res, next) => {
+  let wordcount = req.query.wordcount;
+  fs.readFile('./routes/wordlist.txt', 'utf8', function (err, data) {
+    if (err) {
+      res.status(500).send('Error Occurred');
+      return console.log(err);
+    }
+    data = data.replace(/(?:\r\n|\r|\n)/g, '').replace(/( )/g, '');
+    evalute(data, wordcount, res);
```

Comment:
i am anti sending req/res to any function
makes code more coupled and hard to debug.
instead lets rename `evaluate` to getNextWords set which return a promise
This promise resolves to an object containing next N words data or fails with some error
then you send it back to res(either 200 or 500)